app/actions/transactionHistoryActions.js

```diff
@@ -55,8 +65,11 @@ function parseAbstractData(data, currentUserAddress, tokens) {
     return abstract.address_from
   }
 
-  return data.map(abstract => {
-    const asset = parsedAsset(abstract)
+  const results = []
```

Comment:
I would prefer in general to use reduce instead of creating an array and then pushing things into it from a loop